src/via-structures.stanza

```diff
@@ -0,0 +1,617 @@
+vias in polar space at a particular radius around the signal
+vias at the center of the via structure.
+
+TODO - Diagram Here
+
+Vias start at index 0 which is placed at (radius, 0.0). Vias 1 to `N-1` are
+then place in a counter-clock wise pattern at `radius` from the origin in increments
+of `2 * PI / count` radians.
+
+The `theta` parameter can provide an initial angle offset in degrees. A positive value
+advances the angle such that Via 0 starts at (radius, theta * PI / 180.0)
+
+The `skips` argument provides a means of depopulating particular via indices depending
+on what structure the user wants to build. This collection will contain 0 or more
+zero-based indices for via locations.
+
+<DOC>
+public defstruct PolarViaGroundCage <: ViaGroundCage :
+  doc: \<DOC>
+  Via Definition to use for all of the via placements.
+  <DOC>
+  via-def:Via with:
+    as-method => true
+  doc: \<DOC>
+  Total number of via placements. Must be positive.
+  <DOC>
+  count:Int with:
+    ensure => ensure-positive!
+  doc: \<DOC>
+  Radius in mm for the circular pattern of via placements. Must be positive.
+  <DOC>
+  radius:Double with:
+    ensure => ensure-positive!
+  doc: \<DOC>
+  Starting angle for the pattern. Value in degrees.
+  <DOC>
+  theta:Double with:
+    default => 0.0
+  doc: \<DOC>
+  Skipped indices in the via pattern. Each value in this collection
+  must be in the range `[0, count-1]`
+  <DOC>
+  skips:Collection<Int> with:
+    default => []
+with:
+  constructor => #PolarViaGroundCage
+  printer => true
+
+doc: \<DOC>
+Polar Via Ground Cage Constructor
+
+@param via-def Via to use for all via placements
+@param count Number of via placements
+@param radius Radius in mm of the via placement arc.
+@param theta-deg Initial angle in degrees. Default value is 0.0
+@param skips Collection of via placement indicies to skip. Default value is `[]`
+which implies no placements are skipped.
+<DOC>
+public defn PolarViaGroundCage (
+  --
+  via-def:Via,
+  count:Int,
+  radius:Double,
+  theta-deg:Double = 0.0,
+  skips:Collection<Int> = []
+  ) -> PolarViaGroundCage :
+  ensure-positive!("count", count)
+  ensure-positive!("radius", radius)
+
+  for (skip in skips, i in 0 to false) do:
+    if not (skip < count):
+      throw $ InvalidSkipIndexError("Skip[%_]: %_ < %_" % [i, skip, count])
+
+  #PolarViaGroundCage(via-def, count, radius, theta-deg, skips)
+
+defn compute-pos (v:PolarViaGroundCage, N:Int) -> Pose :
+  val phase = (2.0 * PI * to-double(N) / to-double(count(v))) + to-radians(theta(v))
+  val r = radius(v)
+  loc(r * cos(phase), r * sin(phase))
+
+defmethod make-ground-cage (v:PolarViaGroundCage, n:JITXObject -- pose:Pose) -> False:
+
+  val valid-pos = for i in 0 to count(v) filter:
+    not contains?(skips(v), i)
+
+  inside pcb-module:
+    geom(n):
+      for i in valid-pos do:
+        val pos = pose * compute-pos(v, i)
+        via(via-def(v)) at apply(pos, Point(0.0, 0.0))
```

Comment:
My my info:
is this the same as 
`at Point(0.0, 0.0) * pos`
?

Reply:
Ya - I think this was just a brain fart. I tried using a `Pose` after the `at` and that didn't work - it was looking for a point. I could have done `at pos * Point(0.0, 0.0)`